Documentation/Relationship/Relationship.md

```diff
@@ -0,0 +1,334 @@
+User conforms to the protocol `Identifiable`, so it is also known that every `User` instance is uniquely identified by the
+`id` property.  
+Furthermore the `UserHandler` is registered with a path parameter having the same `User.ID` type, meaning
+we can assume that the `/user/:userId` route can be used to query a certain `User` instance
+(meaning `:userId` translates to `User.id` property).
+
+_Note: As the type of a `@PathParameter` is a `LosslessStringConvertible` it most certainly is either a 
+`String` or an `Int`. Meaning checking if the type of the `User.id` property is equal to `User.ID` type might not be enough.
+Thus it could be necessary for `@PathParemeter`s to specify the type: e.g. `@PathParameter(type: User.self)` where
+the `type` argument must conform to the `Identifiable` protocol (or something similar as `Identifiable.ID` conforms 
+to `Hashable` and we actually need `LosslessStringConvertible`)_
+
+With that information a potential REST exporter could encode the following for a response of the `UserHandler`:
+
+```json
+{
+  "id": "472",
+  "name": "Andreas",
+  "lastname": "Bauer",
+  "email": "mail@example.org",
+  "_links": {
+    "self": "https://example.api/user/472"
+  }
+}
+```
+
+_Note: to be fair the `self` link can also be more easily retrieved by looking at what the client sent in its request.
+However what we actually do is building a mapping from types to their respective `Handler`
+which is needed for the following chapters._
+
+### 3.2. `Handler`s which return an `Identifiable` but don't have a parameter
+
+A special case to [3.1](#31-looking-at-the-return-type-of-handle) is, when we detect that a `Handler` returns a type
+conforming to `Identifiable`, but doesn't have a `@PathParameter` which can be mapped to `.id` property.
+
+This case is illustrated by the following example web service with the `MeUserHandler`:
+```swift
+struct TestService: WebService, Component {
+  @PathParameter var userId: User.ID
+  @PathParameter var postId: User.ID
+
+  var content: some Component {
+    Group("user", $userId) {
+      UserHandler(userId: $userId)
+      // could contain other routes giving more information for the given user, e.g. ./posts
+    }
+    Group("me") {
+      MeUserHandler() // handle() returns `User`
+    }
+  }
+}
+```
+
+In such a case, we can search the web service for a `Handler` which also returns `User`, but has a `@PathParameter`.
+We could then find the `UserHandler` registered under `/user/:userId`.
+
+A response returned from the `MeUserHandler` would then include a `self` relationship pointing to `/user/{userId}` 
+(`userId` is known from the returned `User` instance). More specifically the response will include all relationships
+which are also returned on a request to `user/:userId` (meaning links to potential subroutes).
+
+**Open questions:**
+* This approach heavily relies on the fact that `/me` returns the same type as `user/:userId`. It might be worth
```

Comment:
Maybe we could introduce a concept of a `defaultReference`, e.g.: `defaultReference(to: User.self, id: $userId)` that can be attached to `UserHandler(userId: $userId)` 
```swift
 struct TestService: WebService, Component {
   @PathParameter var userId: User.ID
   @PathParameter var postId: User.ID

   var content: some Component {
     Group("user", $userId) {
       UserHandler(userId: $userId)
         .defaultReference(to: User.self, id: $userId)
       // could contain other routes giving more information for the given user, e.g. ./posts
     }
     Group("me") {
       MeUserHandler() // handle() returns `User`
     }
   }
 }
 ```
We would need to have some startup assertions that there is only one `defaultReference ` but this might solve a few of the problems that are posed by your question.

Reply:
The problem described here was solved with the mechanism described in [3.3.2.2](https://github.com/Apodini/Apodini/blob/feature/realtionship/Documentation/Relationship/Relationship.md#3322-relationship-definition-inherits).

Additionally I took your suggestions of `defaultReference` to solve a different problem [3.2](https://github.com/Apodini/Apodini/blob/feature/realtionship/Documentation/Relationship/Relationship.md#32-using-defaultreference-to-resolve-ambiguous-return-type-information).